### PR TITLE
[Prototype] Reference Http.Features in Http.Abstractions

### PR DESCRIPTION
**Pull request by kichalla**
@Tratcher

### CODE REVIEW

File: src/Microsoft.AspNet.Session/DistributedSession.cs [hunk 1]
```diff
@@ -75,9 +71,7 @@ public void Set(string key, ArraySegment<byte> value)
                 throw new InvalidOperationException("The session cannot be established after the response has started.");
             }
             _isModified = true;
-            byte[] copy = new byte[value.Count];
-            Buffer.BlockCopy(value.Array, value.Offset, copy, 0, value.Count);
-            _store[encodedKey] = copy;
+            _store[encodedKey] = value;
```
Comment (Tratcher) [inline src/Microsoft.AspNet.Session/DistributedSession.cs]: You still need a copy to prevent unintended modification.

**Comment by Tratcher** [comment]
:watch:

**Comment by kichalla** [comment]
Updated. /cc: @Tratcher

**Comment by Tratcher** [comment]
:shipit:

**Comment by kichalla** [comment]
commit dab08ba7e90027a3bf1ef69f740427e93a310f09

**Pull request closed without merging**